### PR TITLE
Added nameable, capitalize, Decorator and a trimmer

### PR DESCRIPTION
**Pull request by stanleySimeon**
# In this milestone I used the Decorator design pattern to correct the names of people
 
  - I created a class Nameable that implements a method called `correct_name` that will raise a `NotImplementedError`.
  - I turned the Person class to Nameable
  - I created a class Decorator and make sure that this is inherited from Nameable
  - I create a class called CapitalizeDecorator and once called TrimmerDecorator and make sure that classes are inherited from the base Decorator class.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+  super initialize(nameable)
+  @nameable = nameable
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+def correct_name
+  @nameable.correct_name
+end
```
Comment (DeliceLydia) [inline decorator.rb]: - [x] Please add `attr_accessor: nameble`  and remember to close all methods well
```
class Decorator < Nameable
    attr_accessor :nameable
  
    def initialize(nameable)
      @nameable = nameable
    end
  

    def correct_name
      @nameable.correct_name
    end
end
    
```

File: person.rb [hunk 2]
```diff
@@ -2,40 +2,41 @@
...
-  capitalizedPerson = CapitalizeDecorator.new(person)
-  capitalizedPerson.correct_name
-  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
-  capitalizedTrimmedPerson.correct_name
\ No newline at end of file
+person.correct_name
+capitalized_person = CapitalizeDecorator.new(person)
+capitalized_person.correct_name
+capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
+capitalized_trimmed_person.correct_name
```
Comment (DeliceLydia) [inline person.rb]: - [x] I am not able to run these codes please address the issue and I suggest that you add in the README how to start the project

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @stanleySimeon, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb (hunk 2 above)
Comment (stanleySimeon) [reply person.rb]: Kindly, can you write more details, please!
I'm not sure I understand what you said

Thanks

Comment (stanleySimeon) [reply person.rb]: @DeliceLydia, what should I do with this pull request that has already been created?

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @stanleySimeon, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failed linters. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by stanleySimeon** [comment]
Hello Lydia
Can you tell me, please what should I do?

**Comment by stanleySimeon** [comment]
Or, What do you think if I leave the project and repeat the week just
because the reviewers won't be able to tell me what I should change when
the project can't be approved?

**Review by KayLemba — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello  @stanleySimeon      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**